// gatsby-plugin-material-ui/gatsby-ssr.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.onRenderBody = exports.wrapRootElement = void 0;

var _react = _interopRequireDefault(require("react"));

var _reactJss = require("react-jss");

var _CssBaseline = _interopRequireDefault(require("@material-ui/core/CssBaseline"));

var _styles = require("@material-ui/core/styles");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; var ownKeys = Object.keys(source); if (typeof Object.getOwnPropertySymbols === 'function') { ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function (sym) { return Object.getOwnPropertyDescriptor(source, sym).enumerable; })); } ownKeys.forEach(function (key) { _defineProperty(target, key, source[key]); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var defaultOptions = {
  theme: {},
  dangerouslyUseGlobalCSS: false,
  productionPrefix: 'jss'
};
var sheetsRegistryMap = new Map();

var wrapRootElement = function wrapRootElement(_ref, options) {
  var element = _ref.element,
      pathname = _ref.pathname;

  var _defaultOptions$optio = _objectSpread({}, defaultOptions, options),
      dangerouslyUseGlobalCSS = _defaultOptions$optio.dangerouslyUseGlobalCSS,
      productionPrefix = _defaultOptions$optio.productionPrefix,
      theme = _defaultOptions$optio.theme;

  var generateClassName = (0, _styles.createGenerateClassName)({
    dangerouslyUseGlobalCSS: dangerouslyUseGlobalCSS,
    productionPrefix: productionPrefix
  });
  var sheetsRegistry = new _reactJss.SheetsRegistry();
  sheetsRegistryMap.set(pathname, sheetsRegistry);
  return _react["default"].createElement(_reactJss.JssProvider, {
    registry: sheetsRegistry,
    generateClassName: generateClassName
  }, _react["default"].createElement(_styles.MuiThemeProvider, {
    theme: (0, _styles.createMuiTheme)(theme),
    sheetsManager: new Map()
  }, _react["default"].createElement(_CssBaseline["default"], null), element));
};

exports.wrapRootElement = wrapRootElement;

var onRenderBody = function onRenderBody(_ref2) {
  var setHeadComponents = _ref2.setHeadComponents,
      pathname = _ref2.pathname;
  var sheetsRegistry = sheetsRegistryMap.get(pathname);

  if (sheetsRegistry) {
    setHeadComponents([_react["default"].createElement("style", {
      id: "server-side-jss",
      key: "server-side-jss",
      dangerouslySetInnerHTML: {
        __html: sheetsRegistry.toString()
      }
    })]);
    sheetsRegistryMap["delete"](pathname);
  }
};

exports.onRenderBody = onRenderBody;
